clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -393,6 +393,7 @@ func (p *pingChecker) run() {
 			success++
 			if success == pingChecks {
 				p.creationTimes.Set(key, phaseName(reachabilityPhase, p.svc.Spec.Type), time.Now())
+				return
```

Comment:
I don't think  that is necessary wrong to assert 10 times, maybe is indeed excessive , but the main problem here is that this doing an exec per command.

Another flaw in this code is that if no `ips`are obtained , this will consider it as a success.

We need to do at least two changes

Reply:
oh, so this never returned before 🤦

Reply:
we should get rid for this lines

for the simple return here:

```
			if success == pingChecks {
				p.creationTimes.Set(key, phaseName(reachabilityPhase, p.svc.Spec.Type), time.Now())
				return
                         }
```

Reply:
agree, is confusing to do the check at the end if we are explicitily setting it here ... unless there are multiple of these running in parlallel? do you know if that is even possible? the same service gets called several times?